src/App.tsx

```diff
@@ -1,10 +1,125 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import cn from 'classnames';
 
 export const App: React.FC = () => {
+  const [todoData, setTodoData] = useState<Todo[]>([]);
+  const [todoList, setTodoList] = useState<Todo[]>(todoData);
+
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const [filterParam, setFilterParam] = useState('all');
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const activeTodos = todoData.filter(todo => !todo.completed).length;
+
+  const isCompletedTodos = todoData.some(todo => todo.completed);
+
+  const isAllTodosCompleted =
+    todoData.length > 0 && todoData.every(todo => todo.completed);
+
+  const isOverlayActive = false; // для оверлея
+
+  const createErrorMessage = (message: string) => {
+    setErrorMessage(message);
+
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodoData)
+      .catch(() => createErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    setTodoList(todoData);
+  }, [todoData]);
+
+  const handleFilter = (filter: string) => {
+    setFilterParam(filter);
+
+    setTodoList(
```

Comment:
rember you previous works const filteredTodos = getPreparedTodos(filter,status) ...